### DISCUSSION TITLE
ValueError: An API token is required to verify contract source code. Visit https://etherscan.io/ to obtain a token, and then store it as the environment variable $ETHERSCAN_TOKEN

### DISCUSSION THREAD

**Discussion by mohzourob** [smartcontractkit/full-blockchain-solidity-course-py#577] [Q&A] [answered]
why this happened?

**Comment by cromewar** [comment, marked as answer]
Hello @dash7ou you need to create a etherscan account and get an API key, than you must add that API key to your .env file. Otherwise the contract will not be verified.

**Reply by mohzourob** [reply]
I know that for mainnet but for kovan ?

**Reply by cromewar** [reply]
Yes sir, in order to verify any contract on you must have an API key, for mainnet and also for all public testnest (you can use the same one on all)

**Reply by mohzourob** [reply]
Oh got it, thx. @cromewar
